### PR TITLE
unit tests for import_job concern, move jobs to /jobs dir

### LINKED ISSUE #65: Testing - Write Unit Tests for ImportJob
Tests need to be written for `ImportJob`, which is a module used by all the CSV import jobs.

**Unit Tests**
The following methods should be unit tested:
- #initialize_processed_file
- #validate_headers
- #import_records

### PR DESCRIPTION
**Pull request by wesleyeewong**
Resolves #65 

### Description

- Went with the `shared_examples_for` for testing concerns as suggested by this SO thread: https://stackoverflow.com/questions/16525222/how-to-test-a-concern-in-rails
- Refactored `import_job.rb` a little to make it more unit testable. More specifically removed a couple of the initializer methods.
- Also moved some of the job classes under the `app/job` dir

### Type of change

* expand test converage

### How Has This Been Tested?

* Ran and made sure complete unit test passed
* Did an upload of the `tagged_animal_assessment` category to make sure they imported without errors

### CODE REVIEW

**Comment by corneliusellen** [comment]
Just saw this, will review this weekend. Thank you for contributing!! :)

File: spec/jobs/tagged_animal_assessment_job_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require "rails_helper"
+
+RSpec.describe TaggedAnimalAssessmentJob do
+  it_behaves_like "import job"
```
Comment (corneliusellen) [inline spec/jobs/tagged_animal_assessment_job_spec.rb]: This is awesome!

**Comment by wesleyeewong** [comment]
Re ran the spec tests to make sure nothing broke. :)

**Comment by corneliusellen** [comment]
@wesleyeewong There are a few merge conflicts from @rafaltrojanowski 's branch I just merged in.

**Comment by rafaltrojanowski** [comment]
@wesleyeewong I believe that's nothing serious. If you need my help, let me know :)

**Comment by corneliusellen** [comment]
bitmoji

**Pull request merged**